client/web/src/marketing/SurveyToast.tsx

```
@@ -66,28 +107,32 @@ export const SurveyToast: React.FunctionComponent<SurveyToastProps> = ({ forceVi
         return null
     }
 
-    return (
-        <Toast
-            title="Tell us what you think"
-            subtitle={
-                <span id="survey-toast-scores">How likely is it that you would recommend Sourcegraph to a friend?</span>
-            }
-            cta={
-                <SurveyRatingRadio
-                    onChange={handleDismiss}
-                    openSurveyInNewTab={true}
-                    ariaLabelledby="survey-toast-scores"
+    switch (activeStep) {
+        case ToastSteps.useCase:
+            return (
+                <SurveyUseCaseForm
+                    handleDone={props => {
+                        setUserFeedback(current => ({ ...current, ...props }))
+                        handleSubmit()
+                        handleContinue()
+                    }}
+                    handleDismiss={handleDismiss}
                 />
-            }
-            footer={
-                <Checkbox
-                    id="survey-toast-refuse"
-                    label="Don't show this again"
-                    checked={shouldPermanentlyDismiss}
-                    onChange={event => setShouldPermanentlyDismiss(event.target.checked)}
+            )
+        case ToastSteps.thankYou:
+            return <SurveySuccess handleDismiss={handleDismiss} />
+        case ToastSteps.rate:
+            return (
+                <SurveyUserRatingForm
+                    onChange={score => setUserFeedback(current => ({ ...current, score }))}
+                    handleDismiss={handleDismiss}
+                    handleContinue={handleContinue}
+                    toggleShouldPermanentlyDismiss={toggleShouldPermanentlyDismiss}
+                    shouldPermanentlyDismiss={shouldPermanentlyDismiss}
+                    toggleErrorMessage={toggleErrorMessage}
                 />
-            }
-            onDismiss={handleDismiss}
-        />
-    )
+            )
+        default:
+            return null
```

Should we throw an error if an unknown `activeStep` is set?